models/refresh_token.go

```diff
@@ -100,6 +100,18 @@ func GetValidChildToken(tx *storage.Connection, token *RefreshToken) (*RefreshTo
 	return refreshToken, nil
 }
 
+func FindTokenBySessionID(tx *storage.Connection, sessionId *uuid.UUID) (*RefreshToken, error) {
+	refreshToken := &RefreshToken{}
+	err := tx.Q().Where("session_id = ? and revoked = false", sessionId).Order("created_at asc").First(refreshToken)
+	if err != nil {
+		if errors.Cause(err) == sql.ErrNoRows {
+			return nil, RefreshTokenNotFoundError{}
+		}
+		return nil, err
+	}
+	return refreshToken, nil
+}
```

Comment:
How does this work with the refresh token reuse interval? Do we need to worry about those?

Reply:
Good point. I think we can drop the `revoked =True` check which should then give the latest token(even if it was refreshed during the session interval and revoked)